packages/cli/lib/pipeline.js

```diff
@@ -2,15 +2,71 @@ import { dirname, resolve } from 'path'
 import fromStream from 'rdf-dataset-ext/fromStream.js'
 import rdf from '@zazuko/env'
 import fromFile from 'rdf-utils-fs/fromFile.js'
+import clownface from 'clownface'
 import findPipeline from '../findPipeline.js'
+import discoverManifests from './discoverManifests.js'
+import ns from './namespaces.js'
+
+const discoverOperations = async () => {
+  const ops = rdf.termMap()
+  for await (const { manifest } of discoverManifests()) {
+    manifest
+      .has(rdf.ns.rdf.type, ns.p.Operation)
+      .forEach(x => { // distinguish EcmaScript and EcmaScriptModule?
+        const link = x.out(ns.code.implementedBy).out(ns.code.link).term
+
+        ops.set(x.term, link)
+      })
+  }
+
+  return ops
+}
+
+export const desugarWith = context => dataset => {
+  const { knownOperations } = context
+  const ptr = clownface({ dataset })
```

Comment:
I recommend using clownface via the environment

```suggestion
  const ptr = rdf.clownface({ dataset })
```